pkg/service/rtcservice.go

```diff
@@ -301,14 +302,6 @@ func (s *RTCService) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 					pLogger.Debugw("sending answer", "answer", m)
 				}
 
-				if pi.ID == "" && cr.InitialResponse.GetJoin() != nil {
-					pi.ID = livekit.ParticipantID(cr.InitialResponse.GetJoin().GetParticipant().GetSid())
-					signalStats = telemetry.NewBytesTrackStats(
-						telemetry.BytesTrackIDForParticipantID(telemetry.BytesTrackTypeSignal, pi.ID),
-						pi.ID,
-						s.telemetry)
-				}
-
```

Comment:
these conditions never change so this code should never run

Reply:
This was done as a safety net I believe. The first response should always be a `JoinResponse`, but there were other messages leap frogging (or `JoinResponse` was missed). But, `cr.InitialResponse` would not have changed. It should have checked again `msg`.

But, the first response should be a `JoinResponse`. JS SDK on client side I believe rejects the connection.

Can we actually add something on server side to ensure that the first response is a `JoinResponse`. Logging an error if not a `JoinResponse` would be good enough for now.

Reply:
I'm not following the comment re: this code should never run. I believe with a new connection `reconnect=false`, the pi.ID would not be set.

Raja is right that as long as reconnect=false, client is expecting a JoinResponse as the initial response. The only time JoinResponse is not sent is during a connection resume `reconnect=true`

Reply:
maybe the condition in the code i deleted was supposed to be using `res`? what i meant was that the condition will never evaluate to true because neither `pi.ID` nor `cr.InitialResponse` ever change.

Reply:
> Was that adding a lot to memory usage and problems with load test?

yes

Reply:
> Can we actually add something on server side to ensure that the first response is a `JoinResponse`. Logging an error if not a `JoinResponse` would be good enough for now.

the channel is ordered now so this shouldn't be a problem anymore. we need to keep this for oss users still using legacy signaling but it shouldn't be a problem once that's deprecated.

Reply:
> maybe the condition in the code i deleted was supposed to be using `res`? what i meant was that the condition will never evaluate to true because neither `pi.ID` nor `cr.InitialResponse` ever change.

Yeah you are right. this is code leftover from a previous refactor. We moved pi.ID and stats init above after we added initial connection retries.